### PR TITLE
Jusjohns toolbaritems demo

### PR DESCRIPTION
**Pull request by profexorgeek**
This PR adds a sample project showing how to use ToolbarItem in XAML and code.

### CODE REVIEW

**Comment by profexorgeek** [comment]
Note, I intended to demonstrate icons in the secondary menu in this sample but they are currently not working correctly. I have filed a bug and will make a note in the docs for this sample.

https://github.com/xamarin/Xamarin.Forms/issues/7007

File: UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos.Android/Properties/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="1" android:versionName="1.0" package="com.xamarin.ToolbarItemDemos" android:installLocation="auto">
+	<uses-sdk android:minSdkVersion="28" />
```
Comment (davidbritch) [inline UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos.Android/Properties/AndroidManifest.xml]: Min shouldn't be 28. I think it should be 19?

File: UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/MainPage.xaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+             xmlns:local="clr-namespace:ToolbarItemDemos"
+             mc:Ignorable="d"
+             x:Class="ToolbarItemDemos.MainPage"
+             Padding="10"
+             Title="ToolbarItem Demos">
+    <TableView Intent="Menu">
+        <TableRoot>
+            <TableSection>
+                <TextCell Text="XAML ToolbarItem"
+                          Detail="Implement a ProgressBar in XAML"
```
Comment (davidbritch) [inline UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/MainPage.xaml]: Update :)

File: UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/MainPage.xaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                          Command="{Binding NavigateCommand}"
+                          CommandParameter="{x:Type local:ToolbarItemXaml}" />
+
+                <TextCell Text="Code ToolbarItem"
+                          Detail="Implement a ProgressBar in code"
```
Comment (davidbritch) [inline UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/MainPage.xaml]: Update :)

File: UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemXaml.xaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+             mc:Ignorable="d"
+             x:Class="ToolbarItemDemos.ToolbarItemXaml"
+             Padding="10"
+             Title="Xaml ToolbarItem Demo">
+<ContentPage.ToolbarItems>
```
Comment (davidbritch) [inline UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemXaml.xaml]: Indentation of this block. Does VS not format on save for XAML? VSMac does.

Similarly, each attribute/property on a new line (VSMac format on save does this).

File: UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemXaml.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
```
Comment (davidbritch) [inline UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemXaml.xaml.cs]: Remove (unused).

File: UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemXaml.xaml (hunk 4 above)
Comment (profexorgeek) [reply UserInterface/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemDemos/ToolbarItemXaml.xaml]: It does not. And when I auto-format it often misaligns properties. Our convention is to align them with the property on the above line. I believe VS for PC aligns them versus traditional code indentation to the nearest "tab".

**Pull request merged**